addons/account_bank_statement_extensions/account_bank_statement.py

```diff
@@ -1,135 +1,91 @@
-        'child_ids': fields.one2many('account.bank.statement.line.global', 'parent_id', 'Child Codes', copy=True),
-        'type': fields.selection([
-            ('iso20022', 'ISO 20022'),
-            ('coda', 'CODA'),
-            ('manual', 'Manual'),
-            ], 'Type', required=True),
-        'amount': fields.float('Amount', digits=0),
-        'bank_statement_line_ids': fields.one2many('account.bank.statement.line', 'globalisation_id', 'Bank Statement Lines'),
-    }
     _rec_name = 'code'
-    _defaults = {
-        'code': lambda s,c,u,ctx={}: s.pool.get('ir.sequence').next_by_code(c, u, 'account.bank.statement.line.global'),
-        'name': '/',
-    }
+
+    name = fields.Char(string='OBI', required=True, default='/', help="Originator to Beneficiary Information")
+    code = fields.Char(size=64, required=True, default=lambda self: self.env['ir.sequence'].next_by_code('account.bank.statement.line.global'))
+    parent_id = fields.Many2one('account.bank.statement.line.global', string='Parent Code', ondelete='cascade')
+    child_ids = fields.One2many('account.bank.statement.line.global', 'parent_id', string='Child Codes', copy=True)
+    type = fields.Selection([('iso20022', 'ISO 20022'), ('coda', 'CODA'), ('manual', 'Manual')], required=True)
+    amount = fields.Float(digits=0)
+    bank_statement_line_ids = fields.One2many('account.bank.statement.line', 'globalisation_id', string='Bank Statement Lines')
+
     _sql_constraints = [
         ('code_uniq', 'unique (code)', 'The code must be unique !'),
     ]
 
-    def name_search(self, cr, user, name, args=None, operator='ilike', context=None, limit=100):
-        if not args:
-            args = []
+    def name_search(self, name, args=None, operator='ilike', limit=100):
+        args = args or []
         ids = []
         if name:
-            ids = self.search(cr, user, [('code', 'ilike', name)] + args, limit=limit)
+            ids = self.search([('code', 'ilike', name)] + args, limit=limit)
             if not ids:
-                ids = self.search(cr, user, [('name', operator, name)] + args, limit=limit)
+                ids = self.search([('name', operator, name)] + args, limit=limit)
             if not ids and len(name.split()) >= 2:
                 #Separating code and name for searching
-                operand1, operand2 = name.split(' ', 1) #name can contain spaces
-                ids = self.search(cr, user, [('code', 'like', operand1), ('name', operator, operand2)] + args, limit=limit)
+                operand1, operand2 = name.split(' ', 1)  # name can contain spaces
+                ids = self.search([('code', 'like', operand1), ('name', operator, operand2)] + args, limit=limit)
         else:
-            ids = self.search(cr, user, args, context=context, limit=limit)
-        return self.name_get(cr, user, ids, context=context)
+            ids = self.search(args, limit=limit)
+        return self.name_get()
```

Comment:
Use recordset return by search method in order to call the 'name_get' method !!!